### DISCUSSION TITLE
ray()->showSelectQueries();

### DISCUSSION THREAD

**Discussion by dillingham** [spatie/laravel-ray#236] [Ideas]
```php
ray()->showSelectQueries();
ray()->showInsertQueries();
ray()->showUpdatetQueries();
ray()->showDeleteQueries();
```

Helpful when test has alot of factories or when targeting a specific query type within a action

**Comment by patinthehat** [comment]
@dillingham I feel like this might be overkill, though I'd consider PRing it - but there's no guarantee that @freekmurze would accept such a PR.  Do you have any thoughts on how your suggestion could be made more succinct?

**Reply by dillingham** [reply]
Understandable. The user experience was just a little less ideal when you have 10 insert queries from factories before the select statement I was aiming to debug. But now I realize I could just show queries after factories 🤦‍♂️ Originally I thought showQueries('SELECT') but that accepts a callable.
